javaslang/src/main/java/javaslang/collection/BitSet.java

```diff
@@ -0,0 +1,1024 @@
+            return remove(currentElement).add(newElement);
+        } else {
+            return this;
+        }
+    }
+
+    @Override
+    default BitSet<T> replaceAll(T currentElement, T newElement) {
+        // a set has only one occurrence
+        return replace(currentElement, newElement);
+    }
+
+    @Override
+    default BitSet<T> retainAll(Iterable<? extends T> elements) {
+        return Collections.retainAll(this, elements);
+    }
+
+    @Override
+    BitSet<T> scan(T zero, BiFunction<? super T, ? super T, ? extends T> operation);
+
+    @Override
+    default <U> Set<U> scanLeft(U zero, BiFunction<? super U, ? super T, ? extends U> operation) {
+        return Collections.scanLeft(this, zero, operation, new java.util.ArrayList<>(), (c, u) -> {
+            c.add(u);
+            return c;
+        }, HashSet::ofAll);
+    }
+
+    @Override
+    default <U> Set<U> scanRight(U zero, BiFunction<? super T, ? super U, ? extends U> operation) {
+        return Collections.scanRight(this, zero, operation, new java.util.ArrayList<>(), (c, u) -> {
+            c.add(u);
+            return c;
+        }, HashSet::ofAll);
+    }
+
+    @Override
+    default Iterator<BitSet<T>> sliding(long size) {
+        return sliding(size, 1);
+    }
+
+    @Override
+    Iterator<BitSet<T>> sliding(long size, long step);
+
+    @Override
+    Tuple2<BitSet<T>, BitSet<T>> span(Predicate<? super T> predicate);
+
+    @Override
+    default BitSet<T> tail() {
+        if (isEmpty()) {
+            throw new UnsupportedOperationException("tail of empty BitSet");
+        } else {
+            return remove(head());
+        }
+    }
+
+    @Override
+    default Option<BitSet<T>> tailOption() {
+        return isEmpty() ? Option.none() : Option.some(tail());
+    }
+
+    @Override
+    BitSet<T> take(long n);
+
+    @Override
+    BitSet<T> takeRight(long n);
+
+    @Override
+    default BitSet<T> takeUntil(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        final BitSet<T> result = takeWhile(predicate.negate());
+        return (result.length() == length()) ? this : result;
+    }
+
+    @Override
+    BitSet<T> takeWhile(Predicate<? super T> predicate);
+
+    @Override
+    default java.util.SortedSet<T> toJavaSet() {
+        return toJavaSet(() -> new java.util.TreeSet<>(comparator()));
+    }
+
+    @Override
+    default BitSet<T> union(Set<? extends T> elements) {
+        Objects.requireNonNull(elements, "elements is null");
+        return addAll(elements);
+    }
+
+    // TODO
+    @Override
+    default <T1, T2> Tuple2<TreeSet<T1>, TreeSet<T2>> unzip(
+            Function<? super T, Tuple2<? extends T1, ? extends T2>> unzipper) {
+        Objects.requireNonNull(unzipper, "unzipper is null");
+        return iterator().unzip(unzipper).map(i1 -> TreeSet.ofAll(naturalComparator(), i1),
+                i2 -> TreeSet.ofAll(naturalComparator(), i2));
+    }
+
+    // TODO
```

Comment:
I forget to remove it and I forget to make this comment:
We know nothing about types &lt;T1, T2, T3&gt; so we can't return tuple of BitSets because we need two conversion functions of every unknown types.